### PR TITLE
enforce max_bytes

### PR DESCRIPTION
**Pull request by evanmcc**
right now we're ignoring this.

### CODE REVIEW

**Comment by evanmcc** [comment]
closes #44

File: src/vg_client.erl [hunk 1]
```diff
@@ -48,10 +48,14 @@ fetch(Topic) when is_binary(Topic) ->
 fetch(Topic, Position) ->
     fetch(Topic, Position, ?TIMEOUT).
 
-fetch(Topic, Position, Timeout) ->
+fetch(Topic, Position, Timeout) when is_integer(Timeout) ->
```
Comment (tsloughter) [inline src/vg_client.erl]: Should we maybe just not have this function at all and require the map?

File: test/topic_SUITE.erl [hunk 2]
```diff
@@ -95,3 +95,17 @@ index_bug(Config) ->
 
     ?assertEqual(199, HWM4),
     ?assertEqual(190, length(Reply4)).
+
+limit(Config) ->
```
Comment (tsloughter) [inline test/topic_SUITE.erl:99]: I'd be interested in a test that also does something like max bytes of 1. I'd worry that the client might break in a case that the max bytes is < the size of one record.

**Review by tsloughter — APPROVED** [review]

File: test/topic_SUITE.erl (hunk 2 above)
Comment (evanmcc) [reply test/topic_SUITE.erl:99]: what would you expect to happen if that was the case? an error? no records?

File: src/vg_client.erl (hunk 1 above)
Comment (evanmcc) [reply src/vg_client.erl]: yeah, probably

File: test/topic_SUITE.erl (hunk 2 above)
Comment (tsloughter) [reply test/topic_SUITE.erl:99]: no records.

**Pull request merged**